gcore/gdal.h

```diff
@@ -1290,6 +1290,79 @@ bool CPL_DLL GDALDatasetSetQueryLoggerFunc(
     GDALDatasetH hDS, GDALQueryLoggerFunc pfnQueryLoggerFunc,
     void *poQueryLoggerArg);
 
+/* ==================================================================== */
+/*      Informational utilities about subdatasets in file names         */
+/* ==================================================================== */
+
+/**
+ *  Opaque type used for the C bindings of the C++ GDALSubdatasetInfo class
+ *  @since GDAL 3.8
+*/
+typedef struct GDALSubdatasetInfo *GDALSubdatasetInfoH;
+
+/**
+ * @brief Returns a new GDALSubdatasetInfo object with methods to extract
+ *        and manipulate subdataset information.
+ *        If the pszFileName argument is not recognized by any driver as
+ *        a subdataset descriptor NULL is returned.
+ *        The returned object must be freed with GDALDestroySubdatasetInfo().
+ * @param pszFileName           File name with subdataset information
+ * @note                        This method does not check if the subdataset actually exists.
+ * @return                      Opaque pointer to a GDALSubdatasetInfo object or NULL if no drivers accepted the file name.
+ * @since                       GDAL 3.8
+ */
+GDALSubdatasetInfoH CPL_DLL CPL_STDCALL
+GDALGetSubdatasetInfo(const char *pszFileName);
+
+/**
+ * @brief GDALSubdatasetInfoGetFileName returns the file path (or base) component of a
+ *        subdataset descriptor effectively stripping the information about the subdataset
+ *        and returning the "parent" dataset descriptor.
+ *        The returned string must be freed with CPLFree().
+ * @param hInfo                 Pointer to GDALSubdatasetInfo object
+ * @param pszFileName           File name with subdataset information
+ * @note                        This method does not check if the subdataset actually exists.
+ * @return                      The original string with the subdataset information removed.
+ * @since                       GDAL 3.8
+ */
+const char CPL_DLL *CPL_STDCALL GDALSubdatasetInfoGetFileName(
```

Comment:
"const char*" is for returns of non-freeable strings

```suggestion
char CPL_DLL *CPL_STDCALL GDALSubdatasetInfoGetFileName(
```